drivers/serial/uart_stm32.c

```diff
@@ -663,10 +670,9 @@ static int uart_stm32_poll_in_visitor(const struct device *dev, void *in, poll_i
 	return 0;
 }
 
-typedef void (*poll_out_fn)(
-	USART_TypeDef *usart, uint16_t out);
+typedef void (*poll_out_fn)(USART_TypeDef *usart, void *out);
 
-static void uart_stm32_poll_out_visitor(const struct device *dev, uint16_t out, poll_out_fn set_fn)
+static void uart_stm32_poll_out_visitor(const struct device *dev, void *out, poll_out_fn set_fn)
```

Comment:
As already mentioned, please limit your changes to RS485 feature.
Other changes should at minimum go in another commit or better a dedicated PR.